rsocket-core/src/main/java/io/rsocket/internal/UnboundedProcessor.java

```diff
@@ -415,7 +420,46 @@ public int requestFusion(int requestedMode) {
 
   @Override
   public void dispose() {
-    cancel();
+    if (cancelled) {
+      return;
+    }
+
+    error = new CancellationException("Disposed");
+    done = true;
+
+    boolean once = true;
+    if (WIP.getAndIncrement(this) == 0) {
+      cancelled = true;
+      int m = 1;
+      for (; ; ) {
+        final CoreSubscriber<? super T> a = this.actual;
+
+        if (!outputFused) {
+          clear();
+        }
+
+        if (a != null && once) {
+          try {
+            System.out.println("sending error");
+            a.onError(error);
+          } catch (Throwable ignored) {
+            System.out.println("bubbled");
+            ignored.printStackTrace();
```

Comment:
Two more lines here for use with a logger?

Reply:
fixed